### PR TITLE
Millionaire support for tracker, via score or clone gun

### PR DESCRIPTION
**Pull request by mauvealerts**
I almost gave up on the score check, tbh. I didn't expect to need to check three numbers, and miscalculated a memory offset at one point.

### CODE REVIEW

File: src/modlunky2/mem/state.py [hunk 1]
```diff
@@ -203,6 +203,14 @@ def hud_flags(self):
         offset = self._offset + 0xA10
         return HudFlags(self._proc.read_u32(offset))
 
+    @property
+    def money_shop_total(self):
```
Comment (gmjosack) [inline src/modlunky2/mem/state.py:207]: do you know if this state counts when you lose money from monkies/leprechauns?

File: src/modlunky2/ui/trackers/runstate.py [hunk 2]
```diff
@@ -526,6 +530,30 @@ def update_is_chain(self):
         if self.win_state is WinState.TIAMAT:
             self.fail_chain()
 
+    def update_millionaire(self, inventory: Inventory):
+        collected = inventory.money + inventory.collected_money_total
```
Comment (gmjosack) [inline src/modlunky2/ui/trackers/runstate.py:534]: I think that these inventory attributes could potentially return None in which case we'd wanna defer on updating here.

File: src/modlunky2/ui/trackers/runstate.py [hunk 3]
```diff
@@ -526,6 +530,30 @@ def update_is_chain(self):
...
+        shop = self.get_critical_state("money_shop_total")
+        self.net_score = collected + shop
+
+        # The category requires completion, which gives at least a $100K bonus.
+        if self.net_score >= 900_000:
```
Comment (gmjosack) [inline src/modlunky2/ui/trackers/runstate.py:539]: oh man, i completely missed that this syntax was added in py3.6. Bless you for bringing it to my attention :D

File: src/modlunky2/mem/state.py (hunk 1 above)
Comment (mauvealerts) [reply src/modlunky2/mem/state.py:207]: I hadn't investigated before, but... yes, leprechauns send you an invoice for the money they stole. Updated docstring

File: src/modlunky2/ui/trackers/runstate.py (hunk 2 above)
Comment (mauvealerts) [reply src/modlunky2/ui/trackers/runstate.py]: I'm unsure, but the checks are done elsewhere. I added them here.

File: src/modlunky2/ui/trackers/runstate.py (hunk 3 above)
Comment (mauvealerts) [reply src/modlunky2/ui/trackers/runstate.py:554]: lol, I wondered whether Python had picked up this syntax I loved in Ruby. Tried and it worked 😸

**Comment by gmjosack** [comment]
This looks good but wanna give rule discussion a couple days to see if anyone has objections

**Pull request merged**